python/services/evaluationservice/dmod/evaluationservice/evaluation_service/consumers/listener.py

```diff
@@ -476,43 +494,66 @@ def scope_data(self) -> ConcreteScope:
 
         return self.__scope
 
-    def receive_subscribed_message(self, message):
+    def receive_subscribed_message(self, message: typing.Union[typing.Dict[str, typing.Any], str, bytes]):
         """
         Interprets and transforms messages sent along the redis channel.
 
         Args:
             message: A message that was published from redis
         """
+        request_id: typing.Optional[str] = None
+
         def is_message_wrapper(possible_wrapper) -> bool:
             return isinstance(possible_wrapper, dict) \
                    and not possible_wrapper.get("event") \
                    and "data" in possible_wrapper
 
+        def get_request_id(container: typing.Union[typing.Dict[str, typing.Any], str, bytes]) -> typing.Optional[str]:
+            if isinstance(container, typing.Mapping) and "request_id" in container:
+                new_request_id = container["request_id"] or request_id
+
+                if isinstance(new_request_id, bytes):
+                    new_request_id = new_request_id.decode()
+
+                return new_request_id
+            return request_id
```

Comment:
This looks like it is intended to return the value from the outer scope in this case, but pointing it out just as a sanity check.

Reply:
yeah. I'm thinking about going back and making it a pure function to force you to input that value, but I'm not sure. The logic for making that check got too repetitive, so that's how I set it up. Not 100% married to the imp, though.